src/main/java/com/derongan/minecraft/guiy/gui/ClickableElement.kt

```diff
@@ -0,0 +1,47 @@
+package com.derongan.minecraft.guiy.gui
+
+import java.util.function.Consumer
+
+
+/**
+ * Wrapper [Element] with a click callback.
+ *
+ * @param wrapped Element to delegate all clicks to.
+ * @param clickAction the action to take when clicked.
+ */
+open class ClickableElement @JvmOverloads constructor(
+        private val wrapped: Element,
+        private var clickAction: (ClickEvent) -> (Unit) = { it.isCancelled = true }
+) : Element {
+    override val size: Size
+        get() = wrapped.size
+
+    override fun draw(guiRenderer: GuiRenderer) {
+        wrapped.draw(guiRenderer)
+    }
+
+    override fun onClick(clickEvent: ClickEvent) {
+        clickAction(clickEvent)
+    }
+
+    /**
+     * Sets the onClick action.
+     *
+     * @param clickAction the action to take when clicked.
+     */
+    fun setClickAction(clickAction: Consumer<ClickEvent>) {
```

Comment:
Is having two really necesary? I would imagine kotlin knows how to coerce into the right functional interface like java does with method refs

Reply:
Using that one in Kotlin requires you to say `button.setClickAction(Consumer { /*some code*/ })` as opposed to `button.setClickActionKt { /*some code*/ }`, which I'd consider to be worse enough to warrant this.